libsolidity/ast/Types.h

```diff
@@ -308,20 +308,67 @@ class IntegerType: public Type
 };
 
 /**
- * Integer constants either literals or computed. Example expressions: 2, 2+10, ~10.
+ * A fixed point type number (signed, unsigned).
+ */
+class FixedType: public Type
+{
+public:
+	enum class Modifier
+	{
+		Unsigned, Signed
+	};
+	virtual Category category() const override { return Category::Fixed; }
+
+	explicit FixedType(int _Lbits, int _Rbits, Modifier _modifier = Modifier::Unsigned);
+
+	virtual bool isImplicitlyConvertibleTo(Type const& _convertTo) const override;
+	virtual bool isExplicitlyConvertibleTo(Type const& _convertTo) const override;
+	virtual TypePointer unaryOperatorResult(Token::Value _operator) const override;
+	virtual TypePointer binaryOperatorResult(Token::Value _operator, TypePointer const& _other) const override;
+
+	virtual bool operator==(Type const& _other) const override;
+
+	virtual unsigned calldataEncodedSize(bool _padded = true) const override { return _padded ? 32 : (m_leftBits + m_rightBits) / 8; }
+	virtual unsigned storageBytes() const override { return (m_leftBits + m_rightBits) / 8; }
+	virtual bool isValueType() const override { return true; }
+
+	virtual std::string toString(bool _short) const override;
+
+	virtual TypePointer encodingType() const override { return shared_from_this(); }
+	virtual TypePointer interfaceType(bool) const override { return shared_from_this(); }
+
+	int numBits() const { return m_leftBits + m_rightBits; }
+	int leftBits() const { return m_leftBits; }
+	int rightBits() const { return m_rightBits; }
+	bool isSigned() const { return m_modifier == Modifier::Signed; }
+
+private:
+	int m_leftBits;
+	int m_rightBits;
+	Modifier m_modifier;
+};
+
+/**
+ * Integer and fixed precision constants either literals or computed. 
+ * Example expressions: 2, 3.14, 2+10.2, ~10.
  * There is one distinct type per value.
  */
-class IntegerConstantType: public Type
+class ConstantNumberType: public Type
 {
 public:
-	virtual Category category() const override { return Category::IntegerConstant; }
+	enum class Modifier
+	{
+		Integer, Fixed
```

Comment:
Do we really need this? I would say `1.0` is an integer, as is the result of `(2.5 * 2)`.